### ISSUE TITLE
Tumour/Normal fields wrong way round for some calls in freebayes output

### ISSUE THREAD

**Issue by phu5ion** [chapmanb/bcbio-nextgen#1848]
Hi,

While parsing a freebayes VCF file, I noticed that the last two fields on tumour/normal depth counts etc seemed to be the wrong way round, for some of the calls. I am sure that this is an error as I concurrently ran mutect and varscan and the calls were alright.
For example, within the same vcf file:
Chr	Pos	RefDepthNormal	AltDepthNormal	VAFNormal	RefDepthTumour	AltDepthTumour	VAFTumour
chr12	25245350	1410	412	0.226	1557	0	0
chr17	7674230	1286	1	0.001	830	1095	0.569

The chr12 variant is definitely the wrong way round and the chr17 variant is correct.
Here's the vcf file, truncated to show only these two calls for brevity. I'm pretty sure it's not an issue with the Freebayes version as this issue didn't crop up when I last ran bcbio on the same Freebayes version.
[faulty_freebayes_output.zip](https://github.com/chapmanb/bcbio-nextgen/files/826446/faulty_freebayes_output.zip)

Thank you!

**Comment by chapmanb** [comment]
Thanks for the report. FreeBayes 1.1.0 did change the output order of multi-sample files. Previously they were in the order of the input on the commandline but that is no longer true with the latest versions. The latest release of bcbio (1.0.2) has fixes to handle this change.

However, I haven't seen inconsistent outputs between different called regions with FreeBayes. We do assume the sample order is consistent across all the regional VCF calls in a run. I'd definitely like to dig into this more and resolve the issue.

Could you provide the original regional VCFs called by FreeBayes for these two samples? These should be in `freebayes/chr12/your_batch-chr12_start_end.vcf.gz`. I'm trying to check if they have the same FreeBayes version and identify what would cause different sample orders. Thanks for the report and help debugging this.

**Comment by phu5ion** [comment]
Hi Brad,

I checked out the regional vcfs and indeed the sample order is inconsistent.
[freebayes.zip](https://github.com/chapmanb/bcbio-nextgen/files/829412/freebayes.zip)
I think I might have restarted variant calling without removing intermediate files once, as I was running on a cluster and my allocated time ran out. Do you think this may have caused the issue?

**Comment by chapmanb** [comment]
Thanks much for the example and confirming the issue. The inconsistent outputs are with the same FreeBayes version and commandline calls. I pushed a fix to the latest development version which will force a consistent output order using `bcftools view -s`. After updating (`bcbio_nextgen.py upgrade -u development`). This will require re-running the freebayes part of your analysis to ensure they are all synchronized. I also filed an issue upstream to hopefully address the underlying issue in FreeBayes as well. Thank you for the help identifying the issue.
